### ISSUE TITLE
Support CRC32C (Castagnoli)

### ISSUE THREAD

**Issue by TattdCodeMonkey** [TattdCodeMonkey/crc#12]
Requested from @fire [here](https://github.com/TattdCodeMonkey/crc/issues/5#issuecomment-337441034)

**Comment by TattdCodeMonkey** [comment]
@fire I'm assuming that is the CRC32_C available here: http://www.sunshine2k.de/coding/javascript/crc/crc_js.html

That is one of the online apps I use for verification when adding new algorithms.

With the addition of CRC32 in #11 I should be able to add the correct polynomials etc. I'll look into it after that is merged.

**Comment by fire** [comment]
The constant 0x1EDC6F41 is what I find when I search for CRC32C. That seems right.

**Comment by potatosalad** [comment]
[pycrc](https://github.com/tpircher/pycrc) also has [crc-32c](https://pycrc.org/models.html#crc-32c) builtin and the table it generates matches the table shown on the [javascript CRC calculator](http://www.sunshine2k.de/coding/javascript/crc/crc_js.html).

You can generate the C source and header with:

```bash
./pycrc.py --generate=c --algorithm=tbl --model=crc-32c
./pycrc.py --generate=h --algorithm=tbl --model=crc-32c
```

**Comment by fire** [comment]
Any progress on this?

**Comment by potatosalad** [comment]
`:crc_32c` is supported under [TattdCodeMonkey/crc@single](https://github.com/TattdCodeMonkey/crc/tree/single) as mentioned in #13

```elixir
iex> CRC.crc(:crc_32c, "123456789")
3808858755
```

**Comment by TattdCodeMonkey** [comment]
What he said, it will be in the next minor release. Currently working through incorporating the TattdCodeMonkey/crc@single branch into a release. But that is behind #15 right now in priority.

**Comment by TattdCodeMonkey** [comment]
@fire 

```elixir
iex> CRC.crc(:crc_32c, "123456789")
3808858755
``` 

Is now available in `v0.8.0` give it a try and let me know if you have any issues
